packages/machine/test/unit/cf-operations/install.spec.ts

```diff
@@ -0,0 +1,121 @@
+import * as cf from "@counterfactual/cf.js";
+import { keccak256 } from "ethers/utils";
+
+import { OpInstall } from "../../../src/middleware/protocol-operation";
+
+import {
+  constructContractCall,
+  constructMultiSend,
+  constructMultiSendSubCall,
+  constructMultisigExecTransaction,
+  TEST_APP_INSTANCE,
+  TEST_FREE_BALANCE,
+  TEST_FREE_BALANCE_APP_INSTANCE,
+  TEST_MULTISIG_ADDRESS,
+  TEST_NETWORK_CONTEXT,
+  TEST_SIGNING_KEYS,
+  TEST_TERMS
+} from "./fixture";
+
+function constructFreeBalanceInput() {
+  const {
+    alice,
+    bob,
+    aliceBalance,
+    bobBalance,
+    localNonce,
+    timeout
+  } = TEST_FREE_BALANCE;
+  const appStateHash = keccak256(
+    cf.utils.abi.encode(
+      ["address", "address", "uint256", "uint256"],
+      [alice, bob, aliceBalance, bobBalance]
+    )
+  );
+  return constructMultiSendSubCall(
+    "delegatecall",
+    TEST_NETWORK_CONTEXT.registryAddr,
+    0,
+    "proxyCall(address,bytes32,bytes)",
+    [
+      TEST_NETWORK_CONTEXT.registryAddr,
+      TEST_FREE_BALANCE_APP_INSTANCE.cfAddress(),
+      constructContractCall(
+        "setState(bytes32,uint256,uint256,bytes)",
+        appStateHash,
+        localNonce,
+        timeout,
+        "0x0"
+      )
+    ]
+  );
+}
+
+function constructConditionalTransferInput(nonceUniqueId: number) {
+  const salt = keccak256(
+    cf.utils.abi.encodePacked(["uint256"], [nonceUniqueId])
+  );
+  const uninstallKey = keccak256(
+    cf.utils.abi.encodePacked(
+      ["address", "uint256", "uint256"],
+      [TEST_MULTISIG_ADDRESS, 0, salt]
+    )
+  );
+  const { assetType, limit, token } = TEST_TERMS;
```

Comment:
Can we pass `TERMS_TERMS` directly into the `constructMultiSendSubCall` below instead of extracting out its fields?

Reply:
No, it expects an array. I prefer it this way